operator/pkg/controller-runtime/log.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controllerruntime
+
+import (
+	"log/slog"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func newLogrFromSlog(logger *slog.Logger) logr.Logger {
+	return logr.New(logSink{logr.FromSlogHandler(logger.Handler()).GetSink()})
+}
+
+type logSink struct{ logr.LogSink }
+
+func (w logSink) Error(err error, msg string, keysAndValues ...any) {
+	if msg == "Reconciler error" && isRetryableError(err) {
+		w.LogSink.Info(0, msg, append(keysAndValues, "err", err)...)
```

Comment:
err ->  error to be consistent with logfields.Error ? Or do you intentionally have another field just in case?

Reply:
Hmm, I was a bit on the fence about this one, as we are not actually consistent. I've currently used `err` to be consistent with how the slog error is rendered by default (it doesn't seem to me that we are overriding it at the moment), but another possibility is definitely using `logfields.Error` to be consistent with that one. Do you have any preference?

Reply:
>  I've currently used err to be consistent with how the slog error is rendered by default (it doesn't seem to me that we are overriding it at the moment)

Interesting 🤔 , I was under the impression that logfields.Error will keep the same output (i.e. slog.Error will only set the level=error).

From the user's perspective, I reckon `error` is preferred as the user's existing log extraction query (e.g. datadog, sumologic, elastic, etc) might not work with `err`.

Reply:
> From the user's perspective, I reckon error is preferred as the user's existing log extraction query (e.g. datadog, sumologic, elastic, etc) might not work with err.

Makes sense, I've updated the PR to use logfields.Error instead, which looks indeed more consistent. 

> Interesting 🤔 , I was under the impression that logfields.Error will keep the same output (i.e. slog.Error will only set the level=error).

Nope, `slog.Error` also takes an error as parameter, and automatically output it as `err=...`. Let me double-check whether we already have an override in place, otherwise I'll open a separate PR for that.

Reply:
sounds good :heavy_check_mark:

Reply:
> Nope, slog.Error also takes an error as parameter, and automatically output it as err=.... Let me double-check whether we already have an override in place, otherwise I'll open a separate PR for that.

Scratch that, I mixed slog and logr. Essentially the fact is that `logr.Error` is actually taking an error as first parameter, and then eventually rendering it as `err=...`: https://github.com/cilium/cilium/blob/cb318104f34421d1c36a6e021c92e91e2b512d3a/vendor/github.com/go-logr/logr/slogsink.go#L91

Reply:
FYI: https://github.com/cilium/cilium/pull/35397